File: TDS_Client/Window/mainNew/src/app/components/mapcreator/models/mapCreateDataDto.ts
import { LanguageEnum } from 'src/app/enums/language.enum';
import { Position4D } from './position4d';
import { Position3D } from './position3d';

export class MapCreateDataDto {
    public Name: string;
    public Type = 0;
    public MinPlayers = 0;
    public MaxPlayers = 999;
    public Description: {
        [key in LanguageEnum]: string
    } = { [7]: "", [9]: "" };
    public TeamSpawns: Position4D[][] = [];
    public MapEdges: Position3D[] = [];
    public BombPlaces: Position3D[] = [];
    public MapCenter: Position3D = new Position3D();
}

File: TDS_Server/Dto/Map/MapCreateDataDto.cs
using System.Collections.Generic;
using TDS_Common.Enum;
using EMapType = TDS_Server.Enum.EMapType;

namespace TDS_Common.Dto.Map
{
    public class MapCreateDataDto
    {
        #nullable disable

        public string Name { get; set; }
        public EMapType Type { get; set; }
        public uint MinPlayers { get; set; }
        public uint MaxPlayers { get; set; }
        public Dictionary<ELanguage, string> Description { get; set; }
        public Position4DDto[][] TeamSpawns { get; set; }
        public Position3DDto[] MapEdges { get; set; }
        public Position3DDto[] BombPlaces { get; set; }
        public Position3DDto MapCenter { get; set; }

        #nullable restore
    }
}
